depthai/plugin.cpp

```diff
@@ -0,0 +1,344 @@
+        
+        #ifndef NDEBUG
+                if(color_go){
+                    rgb_count++;
+                }
+                if(depth_go){
+                    depth_count++;
+                }
+                if(rectifL_go){
+                    left_count++;
+                }
+                if(rectifR_go){
+                    right_count++;
+                }
+        #endif
+
+        if(rectifR_go && rectifL_go && depth_go && color_go) {
+            #ifndef NDEBUG
+                all_count++;
+            #endif
+            auto colorFrame = color->tryGet<dai::ImgFrame>();
+            auto depthFrame = depthQueue->tryGet<dai::ImgFrame>();
+            auto rectifL = rectifLeftQueue->tryGet<dai::ImgFrame>();
+            auto rectifR = rectifRightQueue->tryGet<dai::ImgFrame>();
+            
+            cv::Mat color = cv::Mat(colorFrame->getHeight(), colorFrame->getWidth(), CV_8UC3, colorFrame->getData().data());
+            cv::Mat rgb_out{color.clone()};
+            cv::Mat rectifiedLeftFrame = cv::Mat(rectifL->getHeight(), rectifL->getWidth(), CV_8UC1, rectifL->getData().data());
+            cv::Mat LeftOut{rectifiedLeftFrame.clone()};
+            cv::flip(LeftOut, LeftOut, 1);
+            cv::Mat rectifiedRightFrame = cv::Mat(rectifR->getHeight(), rectifR->getWidth(), CV_8UC1, rectifR->getData().data());
+            cv::Mat RightOut{rectifiedRightFrame.clone()};
+            cv::flip(RightOut, RightOut, 1);
+
+            cv::Mat depth = cv::Mat(depthFrame->getHeight(), depthFrame->getWidth(), CV_16UC1, depthFrame->getData().data());
+            cv::Mat converted_depth;
+            depth.convertTo(converted_depth, CV_32FC1, 1000.f);
+            cam_type_ = cam_type {
+                .img0 = cv::Mat{LeftOut},
+                .img1 = cv::Mat{RightOut},
+                .rgb = cv::Mat{rgb_out},
+                .depth = cv::Mat{converted_depth},
+                .iteration = iteration_cam,
+            };
+            iteration_cam++;
+        }
+        // Images
+        std::optional<cv::Mat> img0 = std::nullopt;
+        std::optional<cv::Mat> img1 = std::nullopt;
+        std::optional<cv::Mat> rgb = std::nullopt;
+        std::optional<cv::Mat> depth = std::nullopt;
+
+            
+        if (last_iteration_cam != cam_type_.iteration)
+        {
+            last_iteration_cam = cam_type_.iteration;
+            img0 = cam_type_.img0;
+            img1 = cam_type_.img1;
+            rgb = cam_type_.rgb;
+            depth = cam_type_.depth;
+        }
+        
+        std::chrono::time_point<std::chrono::steady_clock, std::chrono::steady_clock::duration> gyroTs;
+        Eigen::Vector3f la;
+        Eigen::Vector3f av;
+        
+        if(imuPacket_go) {
+            auto imuPacket = imuQueue->tryGet<dai::IMUData>();
+            #ifndef NDEBUG
+                if(imu_packet == 0){
+                    first_packet = std::chrono::steady_clock::now();
```

Comment:
```suggestion
                    first_packet_time = std::chrono::steady_clock::now();
```
A little clearer

Reply:
If #250 gets merged first, change this to be based on the relative clock.

Reply:
I'm thinking this will need some conversion then since the DepthAI library uses steady_clock, but I'm not sure